vars/sectionCI.groovy

```diff
@@ -27,19 +27,21 @@ def call(params) {
   def product = params.product
   def component = params.component
   def aksUrl
+  def environment = params.environment;
+  def parentEnvironment = params.parentEnvironment; // Used for secrets which are maintained only in parent environments.
 
   Builder builder = pipelineType.builder
 
   if (config.dockerBuild) {
-    withAksClient(subscription) {
+    withAksClient(subscription,environment) {
       def acr = new Acr(this, subscription, env.REGISTRY_NAME, env.REGISTRY_RESOURCE_GROUP)
       def dockerImage = new DockerImage(product, component, acr, new ProjectBranch(env.BRANCH_NAME).imageTag(), env.GIT_COMMIT)
 
       onPR {
         acr.retagForStage(DockerImage.DeploymentStage.PR, dockerImage)
 
         if (config.deployToAKS) {
-          withTeamSecrets(config, params.environment) {
+          withTeamSecrets(config, parentEnvironment) {
```

Comment:
`withTeamSecrets` has built in support for overriding vaults between environments https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/withTeamSecrets.groovy#L46.

The `config` object contains a map of environments (configurable by teams) to override the environment to use for vaults. The map defaults to empty `[:]` (i.e. no overrides) but we could default it to `['preview':'aat']`